### PR TITLE
Add gatk/createsequencedictionary

### PR DESCRIPTION
**Pull request by MaxUlysse**

### CODE REVIEW

**Comment by MaxUlysse** [comment]
remove the cpus, memory and time requirements I added to the config files, but I still think it's a good practice to have that especially to limit resources for our CI tests here with GitHub actions

File: software/gatk/createsequencedictionary/main.nf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    if (!task.memory) {
+        log.info '[GATK] Available memory not known - defaulting to 6GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = task.memory.giga
+    }
+    """
+    gatk --java-options "-Xmx${avail_mem}g" \\
+        CreateSequenceDictionary \\
+        --REFERENCE $fasta \\
+        --URI $fasta
```
Comment (drpatelh) [inline software/gatk/createsequencedictionary/main.nf:40]:
```suggestion
        --URI $fasta \\
```

File: software/gatk/createsequencedictionary/meta.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        description: |
+            Instead of directly downloading Singularity images for use with Singularity,
+            force the workflow to pull and convert Docker containers instead.
+input:
+    - fasta:
+        type: file
+        description: Input fasta file
+        pattern: "*.{fasta,fa}"
+output:
+    - bam:
```
Comment (drpatelh) [inline software/gatk/createsequencedictionary/meta.yml:42]:
```suggestion
    - dict:
```

**Review by drpatelh — APPROVED** [review]
Thanks @MaxUlysse !

**Pull request merged**